### DISCUSSION TITLE
uftpd equivalent for Pi Pico W?

### DISCUSSION THREAD

**Discussion by ESPWarren** [micropython/micropython#9575] [Using MicroPython]
Hi.

Is there a lightweight ftp server available that works on a Pi Pico W?

On the ESP32, uftpd was really useful - is there an equivalent for the Pico?

**Comment by robert-hh** [comment]
uftpd.py works on the Pi Pico W as well. I just tried it.

**Reply by ESPWarren** [reply]
Thanks for the confirmation.

Excuse the noob type question, but how did you install uftpd - did you use upip? I have tried that but keep getting 'syntax error' type messages...

Thanks

**Comment by robert-hh** [comment]
I use mpremote.py to copy uftpd.py to the board over USB.

**Comment by ESPWarren** [comment]
Presumably that is identical to using Thonny to copy the uftpd.py library to the device - yes?

**Reply by robert-hh** [reply]
Using thonny should be fine as well. You just have to put that single file to the device.

**Comment by ESPWarren** [comment]
Got it - I had forgotten that import uftpd starts the server - so I was later adding uftpd.start()!!

It has been four years since I used it -such a useful library - thanks
